Tasks/UsePythonVersion/Tests/L0.ts

```diff
@@ -0,0 +1,122 @@
+import * as assert from 'assert';
+import { EOL } from 'os';
+import * as path from 'path';
+import * as mockery from 'mockery';
+import * as usePythonVersion from '../usepythonversion';
+
+/** Reload the unit under test to use mocks that have been registered. */
+function reload(): typeof usePythonVersion {
+    return require('../usepythonversion');
+}
+
+describe('UsePythonVersion L0 Suite', function () {
+    before(function () {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    after(function () {
+        mockery.disable();
+    });
+
+    beforeEach(function () {
+        mockery.registerSubstitute('vsts-task-lib/task', 'vsts-task-lib/mock-task');
+    });
+
+    afterEach(function () {
+        mockery.deregisterAll();
+        mockery.resetCache();
+    })
+
+    it('finds version in cache', async function () {
+        const toolPath = path.join('/', 'Python', '3.6.4');
+        mockery.registerMock('vsts-task-tool-lib/tool', {
+            findLocalTool: () => toolPath
+        });
+
+        const uut = reload();
+        const parameters = {
+            versionSpec: '3.6',
+            outputVariable: 'Python',
+            addToPath: false
+        };
+
+        assert.notEqual(process.env['PYTHON'], toolPath);
```

Comment:
@ericsciple This is failing the second time it runs during the PR builds.  (The VSTS builds first run the tests on Node 6 and then on Node 5.)  `setVariable` from the task lib is not mocked and is really changing the agent's environment.

What we want here is to use a test spy to make sure that `setVariable` is called and with the correct arguments.  Same with tool lib's `prependPath`, which I've mocked ad-hoc below.

Sinon.JS looks like a good choice for mocks and spies.  Do you have any objections to starting to use this for our L0 testing?

Reply:
@brcrista - I would like to avoid a global change like that right now.  is there another solution we could use?  @ericsciple ??

Reply:
I can just mock this ad-hoc again for now.  But using a mocking / spying library will be a comprehensive solution for this, and I think it's something we should consider.